app/actions/local/systems.ts

```diff
@@ -309,3 +322,41 @@ export async function dismissAnnouncement(serverUrl: string, announcementText: s
         return {error};
     }
 }
+
+export async function expiredBoRPostCleanup() {
+    const {database: appDatabase} = DatabaseManager.getAppDatabaseAndOperator();
+    const lastRunAt = await getLastBoRPostCleanupRun(appDatabase);
+    const shouldRunNow = (lastRunAt - Date.now()) > BOR_POST_CLEANUP_MIN_RUN_INTERVAL;
+
+    if (!shouldRunNow) {
+        return {error: undefined};
+    }
+
+    const servers = await getAllServers();
+    for (const server of servers) {
+        const serverDatabase = DatabaseManager.serverDatabases[server.url]?.database;
+        if (!serverDatabase) {
+            continue;
+        }
+
+        /* eslint-disable-next-line no-await-in-loop */
+        await removeExpiredBoRPosts(server.url, serverDatabase);
+    }
+
+    return {error: undefined};
+}
+
+async function removeExpiredBoRPosts(serverUrl: string, database: Database) {
+    try {
+        const allBoRPosts = await queryPostsByType(database, PostTypes.BURN_ON_READ).fetch();
+        const expiredBoRPostIDs = allBoRPosts.
+            filter((post) => isExpiredBoRPost(post)).
+            map((post) => post.id);
+
+        await dataRetentionCleanPosts(serverUrl, expiredBoRPostIDs);
```

Comment:
probably should be called if expiredBoRPostIDs contains anything and is not an empty array